### DISCUSSION TITLE
Sensors for current activity

### DISCUSSION THREAD

**Discussion by bstempien** [sebr/bhyve-home-assistant#151] [General]
Just started using this integration with my Orbit WT24—0001 8 Zone sprinkler controller.
I don't know if this is a feature request or I am missing something, but I don't see any sensors that would tell me that either a particular zone or even just watering in general is currently happening.
What I in particular need is just a binary sensor that basically represents that any zone is currently running. In may case I am reading water consumption from my utility meter and if I see a large volume in a short time period I want to check to see if it is my sprinklers running. If not it could be a water line break.
A sensor that returns what zone is running would also work.

Thanks in advance!

**Comment by pjv** [comment]
I have a different type of b-hyve device (a 2-zone hose timer), so I’m not sure this applies to your device, but for mine, the integration creates a switch entity for each of my two zones and they act like any switch in HA; their state is **on** when the zone is switched on and **off** when it’s switched off.

**Comment by bstempien** [comment]
Each zone does have a switch. What I am seeing is that they are not
updating when a Zone is triggered by a Program or even manual start in the
Orbit App.
Clicking on a switch in the dashboard shows a history where the switch has
been off continuously. The other attributes of the zone do appear to be
correct (what programs they are on, etc.)

The history sensor does update with last run time. But that happens I
believe after the Zone has finished a run.

**Reply by pjv** [reply]
Yes, the history sensor does update after the fact.

You are probably right that the switches don’t update when the device gets turned on and off via the b-hyve app or an automatic program. This is probably along the same lines as my related (unanswered) question #138.

At the moment, I’m only turning on and off the device via Home Assistant in node red flows or automations and so the switches work for me in that context.

**Comment by bstempien** [comment]
Thanks for confirming. I will log this as an issue then.
